doc/hardware/porting/board_porting.rst

```diff
@@ -912,9 +912,67 @@ There are some extra things you'll need to do:
 #. Prepare a pull request adding your board which follows the
    :ref:`contribute_guidelines`.
 
+.. _extend-board:
+
 Board extensions
 ****************
 
+The board hardware model in Zephyr allows you to extend an existing board with
+new board variants. Such board extensions can be done in your custom repository
+and thus outside of the Zephyr repository.
+
+Extending an existing board with an extra variant allows you to adjust an
+existing board and thereby during build to select building for the existing,
+unmodified board, or the new variant.
+
+To extend an existing board, first create a :file:`board.yml` in your extended
+board. Make sure to use the directory structure described in
+:ref:`create-your-board-directory`.
+
+The skeleton of the board YAML file for extending a board is:
+
+.. code-block:: yaml
+
+   board:
+     extend: <existing-board-name>
+      variants:
+       - name: <new-variant>
+         qualifier: <existing-qualifier>
+
+When extending a board, your board directory should look like:
+
+.. code-block:: none
+
+   boards/<VENDOR>/plank
+   ├── board.yml
+   ├── plank_<new-qualifiers>_defconfig
+   └── plank_<new-qualifiers>.dts
+
+Replace ``plank`` with the real name of the board you extend.
+
+In some cases you might want to also adjust additional settings, like the
+:file:`Kconfig.defconfig` or :file:`Kconfig.{board}`.
+Therefore it is also possible to provide the following in addition when
+extending a board.
+
+.. code-block:: none
+
+   boards/<VENDOR>/plank
+   ├── board.cmake
+   ├── Kconfig
+   ├── Kconfig.plank
+   ├── Kconfig.defconfig
+   └── plank_<new-qualifiers>.yaml
```

Comment:
had to scratch my head a bit here trying to understand what the new_qualifier would end up like by looking at the tests. It will help alot if you provide an example here showing how you get to the new qualifier by extending a board.

Reply:
maybe:

As an example, when extending the native_sim board using the following:

```
board:
  extend: native_sim
  variants:
    - name: one
      qualifier: native
    - name: two
      qualifier: native/64
```


the new board targets will end up as `native_sim/native/one` and `native_sim/native/64/two`. The DTS files for the extended boards will be `native_sim_native_64_two.dts` and `native_sim_native_one.dts`.